geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/ConnectionManagerImpl.java

```diff
@@ -86,17 +87,15 @@
   private final InternalLogWriter securityLogWriter;
   protected final CancelCriterion cancelCriterion;
 
-  protected volatile int connectionCount;
+  protected final AtomicInteger connectionCount = new AtomicInteger(0);
   protected ScheduledExecutorService backgroundProcessor;
   protected ScheduledExecutorService loadConditioningProcessor;
 
-  protected ReentrantLock lock = new ReentrantLock();
-  protected Condition freeConnection = lock.newCondition();
   private ConnectionFactory connectionFactory;
   protected boolean haveIdleExpireConnectionsTask;
   protected boolean havePrefillTask;
```

Comment:
havePrefillTask is used by startBackgroundPrefill which thinks it is always called while the "lock" is held.
To prevent multiple prefill tasks from getting created I think we need to make this atomic.

Reply:
Yes